backend/services/implementations/courseService.ts

```diff
@@ -34,7 +35,48 @@ const publishedModulesOnlyQuery = [
 ];
 
 class CourseService implements ICourseService {
+  imageService: IImageUploadService;
+
+  constructor(imageService: IImageUploadService) {
+    this.imageService = imageService;
+  }
+
   /* eslint-disable class-methods-use-this */
+  private async hydrateImages(
```

Comment:
Just so I understand, what is the point of the backend hydration? Are you downloading the image in the backend so all the frontend has to do is use the attribute?

Reply:
Kind of. Firebase uses temporary signed URLs based on a "file path" to access images, but these expire every hour so we can't store them. Each time we want to access an image, we have to recreate this URL (or use one from within the hour). So it's not downloading the image so much as getting the image URL.